### PR TITLE
Add parameter name to validation errors

### PR DESCRIPTION
**Pull request by bernt-matthias**
This could be useful for tool testing. So far it was unclear which parameter had the validation error.

Not sure if this is also user facing .. then the parameter name might not be helpful.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by bernt-matthias** [comment]
> The unit test (actually doc test) failures in `lib/galaxy/tools/parameters/validation.py` are relevant, the others should be fixed by #15441 .

Thanks for checking. Should be easy to fix. Do you think that these should be "proper" unit tests? I created those back in the days when I knew even less about testing than today :)

**Comment by nsoranzo** [comment]
It'd be probably better but I don't have strong feelings on that.

File: lib/galaxy/tools/parameters/basic.py [hunk 1]
```diff
@@ -290,7 +290,10 @@ def validate(self, value, trans=None):
         if value in ["", None] and self.optional:
             return
         for validator in self.validators:
-            validator.validate(value, trans)
+            try:
+                validator.validate(value, trans)
+            except ValueError as e:
+                raise ValueError(f"Parameter {self.name}: {e}") from None
```
Comment (bernt-matthias) [inline lib/galaxy/tools/parameters/basic.py]:
```suggestion
                raise ValueError(f"Parameter {self.name}: {e}") from None
```

File: lib/galaxy/tools/parameters/basic.py [hunk 2]
```diff
@@ -1990,7 +1993,11 @@ def do_validate(v):
                 ):
                     return
                 else:
-                    validator.validate(v, trans)
+                    try:
+                        validator.validate(v, trans)
+                    except ValueError as e:
+                        raise ValueError(f"Parameter {self.name}: {e}") from None
+
```
Comment (bernt-matthias) [inline lib/galaxy/tools/parameters/basic.py:1998]:
```suggestion
```

File: lib/galaxy/tools/parameters/validation.py [hunk 3]
```diff
@@ -15,12 +15,6 @@
 log = logging.getLogger(__name__)
 
 
-def get_test_fname(fname):
-    """Returns test data filename"""
-    path, name = os.path.split(__file__)
-    full_path = os.path.join(path, "test", fname)
-    return full_path
-
```
Comment (bernt-matthias) [inline lib/galaxy/tools/parameters/validation.py:24]:
```suggestion

```

**Comment by bernt-matthias** [comment]
Good to go from my side.

**Comment by mvdbeek** [comment]
Thanks, really appreciate moving the tests to proper unit tests.

**Pull request merged**